From f1fd21675bcfe713889a162540bbc3cd8ab226c0 Mon Sep 17 00:00:00 2001
From: Kevin O'Rourke <misc1-github@caboose.org.uk>
Date: Sun, 31 Jan 2021 11:35:10 +0100
Subject: [PATCH] Fix build problems in Yocto

---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9566b48..7610ec3 100644
--- a/Makefile
+++ b/Makefile
@@ -1716,14 +1716,14 @@ export CONFIG_RTL8189FS = m
 all: modules
 
 modules:
-	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) M=$(shell pwd)  modules
+	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KERNEL_SRC) M=$(shell pwd)  modules
 
 strip:
 	$(CROSS_COMPILE)strip $(MODULE_NAME).ko --strip-unneeded
 
 install:
-	install -p -m 644 $(MODULE_NAME).ko  $(MODDESTDIR)
-	/sbin/depmod -a ${KVER}
+	install -p -m 644 $(MODULE_NAME).ko ${MODLIB}/kernel/drivers/net/wireless
+	#/sbin/depmod -a ${KVER}
 
 uninstall:
 	rm -f $(MODDESTDIR)/$(MODULE_NAME).ko
